lib/tagexts.c

```diff
@@ -742,17 +742,24 @@ static int headercolorTag(Header h, rpmtd td, headerGetFlags hgflags)
 
 static int archsuffixTag(Header h, rpmtd td, headerGetFlags hgflags)
 {
+    const char *a = NULL;
     char * val;
 
     if (headerIsSource(h)) {
 	if (headerIsEntry(h, RPMTAG_NOSOURCE) ||
 	    headerIsEntry(h, RPMTAG_NOPATCH))
-	    val = xstrdup("nosrc");
+	    a = xstrdup("nosrc");
```

Comment:
I think you forgot to delete the `xstrdup` here and below.

Reply:
Doh... thanks for spotting. Fixed in 2cf26bdd2f15e25a7c1a82d998e3fcb8d2669b03